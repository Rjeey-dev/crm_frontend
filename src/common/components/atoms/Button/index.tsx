import React, { ReactNode } from 'react';

interface IProps {
    disabled: boolean,
    classes?: string,
    children?: ReactNode,
    handleClick?: (event: React.MouseEvent<HTMLButtonElement>) => void,
    attributes?: object,
    type?: any
}

function Button(props: IProps) {
    return <button disabled={props.disabled} type={props.type} className={props.classes} onClick={props.handleClick} {...props.attributes}>
            {props.children}
        </button>;
}

Button.defaultProps = {
    disabled: false
};

export default Button;