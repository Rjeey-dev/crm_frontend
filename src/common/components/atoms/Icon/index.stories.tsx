import React from 'react';
import { storiesOf } from '@storybook/react';
import Icon from './index';

storiesOf('Icon', module)
    .add('notifications', () => (
        <Icon classes='mdi mdi-file-document-box'/>
    ))
    .add('settings', () => (
        <Icon classes='mdi mdi-tune'/>
    ));