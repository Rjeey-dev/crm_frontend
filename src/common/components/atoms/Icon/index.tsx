import React from 'react';

interface IProps {
    classes: string,
    attributes?: object
}

function Icon(props: IProps) {
    return <i className={props.classes} {...props.attributes}/>;
}

export default Icon;
