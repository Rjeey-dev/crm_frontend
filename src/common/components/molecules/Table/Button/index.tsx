import IconButton from "@material-ui/core/IconButton";
import { withStyles } from "@material-ui/core/styles";
import Tooltip from "@material-ui/core/Tooltip";
import AddIcon from "@material-ui/icons/Add";
import React from "react";
import {TranslateProps} from "react-localize-redux";

import Translation from "common/components/atoms/Translation/index";

interface IClasses {
    iconButton: string,
    deleteIcon: string
}

interface IProps {
    title: TranslateProps,
    classes: IClasses,
    handleClick?: (event: React.MouseEvent<HTMLButtonElement>) => void,
}

class Button extends React.Component<IProps> {
    public render() {
        return (
            <>
                <Tooltip title={<Translation source={this.props.title}/>}>
                    <IconButton className={this.props.classes.iconButton} onClick={this.props.handleClick}>
                        <AddIcon className={this.props.classes.deleteIcon} />
                    </IconButton>
                </Tooltip>
            </>
        );
    }
}

export default withStyles({}, { name: "CustomToolbar" })(Button);
