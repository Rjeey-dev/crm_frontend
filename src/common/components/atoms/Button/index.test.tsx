import { shallow } from 'enzyme';
import React from 'react';

import Icon from '../Icon/index';
import Button from './index';

it("Renders <Button/> with correct properties", () => {
    const output = shallow(
        <Button disabled={true} classes='class'>
            <Icon classes='icon'/>
            Text
        </Button>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <Button/> with only required props", () => {
    const output = shallow(
        <Button/>
    ).html();

    expect(output).toMatchSnapshot();
});
