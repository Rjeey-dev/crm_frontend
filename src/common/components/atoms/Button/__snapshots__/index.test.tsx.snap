// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Button/> with correct properties 1`] = `"<button disabled=\\"\\" class=\\"class\\"><i class=\\"icon\\"></i>Text</button>"`;

exports[`Renders <Button/> with only required props 1`] = `"<button></button>"`;
